### PR TITLE
fix: parse proto packages without namespace correctly

### LINKED ISSUE #201: service name & version parse incorrectly
For API like `grafeas` which is in `googleapis/gefeas/v1` instead of normal path `googelapis/google/cloud/serviceName/version/protoFile`, service name and version can not be got correctly. For `grafeas` specifically, service name is `' '` and version is undefined.

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**
fix #201

### CODE REVIEW

File: typescript/src/schema/naming.ts [hunk 1]
```diff
@@ -38,8 +38,13 @@ export class Naming {
 
     // Define the regular expression to match a version component
     // (e.g. "v1", "v1beta4", etc.).
-    const pattern = /^((?:[a-z0-9_.]+?)\.)?([a-z0-9_]+)(?:\.(v[0-9]+(p[0-9]+)?((alpha|beta)[0-9]+)?[^.]*))?$/;
-    const match = rootPackage.match(pattern);
+    const pattern1 = /^((?:[a-z0-9_.]+?)\.)?([a-z0-9_]+)(?:\.(v[0-9]+(p[0-9]+)?((alpha|beta)[0-9]+)?[^.]*))?$/;
+    // Special pattern match for API like grafeas which in 'googleapis/grafeas/v1' without namespace.
+    const pattern2 = /^(([a-z0-9_.]+?)\.)((v[0-9]+(p[0-9]+)?((alpha|beta)[0-9]+)?[^.]*))?$/;
+    let match = rootPackage.match(pattern1);
+    if (!rootPackage.startsWith('google')) {
```
Comment (alexander-fenster) [inline typescript/src/schema/naming.ts]: `google.` to make sure the dot is there?

File: typescript/src/schema/naming.ts [hunk 2]
```diff
@@ -35,20 +35,18 @@ export class Naming {
       throw new Error('Protos provided have different proto packages.');
     }
     const rootPackage = prefix.replace(/\.$/, '');
-
-    // Define the regular expression to match a version component
-    // (e.g. "v1", "v1beta4", etc.).
-    const pattern = /^((?:[a-z0-9_.]+?)\.)?([a-z0-9_]+)(?:\.(v[0-9]+(p[0-9]+)?((alpha|beta)[0-9]+)?[^.]*))?$/;
-    const match = rootPackage.match(pattern);
-    if (!match) {
+    const match = rootPackage.split('.');
```
Comment (alexander-fenster) [inline typescript/src/schema/naming.ts]: This is not a `match` anymore. This is... `segments`?

File: typescript/src/schema/naming.ts [hunk 3]
```diff
@@ -35,20 +35,18 @@ export class Naming {
...
-    const [, namespaces, name, version] = match;
-    if (!namespaces) {
-      throw new Error(
-        `Cannot parse package name ${rootPackage}: namespace is not defined.`
-      );
+    const version = match[match.length - 1];
+    // version should follow the pattern of 'v1' or 'v1alpha1'
+    const versionpattern = /^((v[0-9]+(p[0-9]+)?((alpha|beta)[0-9]+)?[^.]*))?$/;
+    if (!version.match(versionpattern)) {
+      throw new Error(`Cannot parse package name ${rootPackage}.`);
```
Comment (alexander-fenster) [inline typescript/src/schema/naming.ts]: `Cannot parse package name ${rootPackage}: version does not match ${versionPattern}.` Error messages must explain what went wrong.

File: typescript/src/schema/naming.ts [hunk 4]
```diff
@@ -35,20 +35,18 @@ export class Naming {
...
       throw new Error(`Cannot parse package name ${rootPackage}.`);
     }
-    const [, namespaces, name, version] = match;
-    if (!namespaces) {
-      throw new Error(
-        `Cannot parse package name ${rootPackage}: namespace is not defined.`
-      );
+    const version = match[match.length - 1];
+    // version should follow the pattern of 'v1' or 'v1alpha1'
+    const versionpattern = /^((v[0-9]+(p[0-9]+)?((alpha|beta)[0-9]+)?[^.]*))?$/;
```
Comment (alexander-fenster) [inline typescript/src/schema/naming.ts]: `versionPattern` (camel case)

File: typescript/src/schema/naming.ts [hunk 5]
```diff
@@ -35,20 +35,18 @@ export class Naming {
...
+    if (!version.match(versionpattern)) {
+      throw new Error(`Cannot parse package name ${rootPackage}.`);
     }
+    const name = match[match.length - 2];
+    const namespaces = match.slice(0, match.length - 2).join('.');
```
Comment (alexander-fenster) [inline typescript/src/schema/naming.ts]: Just `slice(0, -2)`.

```
> [1,2,3,4,5,6,7].slice(0, -2)
[ 1, 2, 3, 4, 5 ]
```

Comment (xiaozhenliu-gg5) [reply typescript/src/schema/naming.ts]: Nice! thank you!

**Pull request merged**